src/app/cli/src/init/client.ml

```diff
@@ -549,6 +549,65 @@ let send_payment_graphql =
          printf "Dispatched payment with ID %s\n"
            ((response#sendPayment)#payment |> unwrap_user_command)#id ))
 
+let sign_payment =
+  let open Command.Param in
+  let open Cli_lib.Arg_type in
+  let receiver_flag =
+    flag "--receiver" ~aliases:["receiver"]
+      ~doc:"PUBLICKEY Public key to which you want to send money"
+      (required public_key_compressed)
+  in
+  let amount_flag =
+    flag "--amount" ~aliases:["amount"]
+      ~doc:"VALUE Payment amount you want to send" (required txn_amount)
+  in
+  let token_flag =
+    flag "--token" ~aliases:["token"]
+      ~doc:"TOKEN_ID The ID of the token to transfer" (optional token_id)
+  in
+  let args =
+    Args.zip4 Cli_lib.Flag.signed_command_common receiver_flag amount_flag
+      token_flag
+  in
+  Command.async ~summary:"generate a signed payment to an address"
+    (Command.Param.map args
+       ~f:(fun (common, receiver_pk, amount, token_id) () ->
+            let {Cli_lib.Flag.sender; fee; nonce; memo} = common in
+            let fee_token = Token_id.default in
+            let fee_payer_pk = sender in
+            let valid_until = Unsigned.UInt32.max_int in
+            let nonce = Option.value ~default:Unsigned.UInt32.zero nonce in
+            let memo = Option.map ~f:Signed_command_memo.of_string memo
+                        |> Option.value ~default:Signed_command_memo.empty in
+            let common: Signed_command_payload.Common.t =
+              { fee; fee_token; fee_payer_pk; nonce; valid_until; memo } in
+            let body: Signed_command_payload.Body.t =
+              let token_id = Option.value ~default:Token_id.default token_id in
+              Signed_command_payload.Body.Payment
+              Payment_payload.Poly.
+              { source_pk=sender; receiver_pk; token_id; amount } in
+            let payload = Signed_command_payload.Poly.{ common; body } in
+            let logger = Logger.create () in
+            let%bind wallets =
+              let conf_dir = Mina_lib.Conf_dir.compute_conf_dir None in
+              let wallets_disk_location = conf_dir ^/ "wallets" in
+              Secrets.Wallets.load ~logger
+                ~disk_location:wallets_disk_location
+            in
+            let sk_file = Secrets.Wallets.get_path wallets sender in
+            let%bind kp =
+              Secrets.Keypair.Terminal_stdin.read_from_env_exn ~logger
+                ~which:"block producer keypair" sk_file
+            in
+            let sk = kp.private_key in
+            let signature = Signed_command.sign_payload sk payload in
+            let signer: Public_key.t = Public_key.decompress_exn sender in
+            let signed: Signed_command.t = Signed_command.Poly.{ payload; signer; signature } in
+            Signed_command.to_string signed
```

Comment:
Can you document which format this is using in the command description, and perhaps speak a little to why you chose this format rather than an alternative? Please also add a transcript from a test run of the command as a comment on the PR.

Reply:
>Can you document which format this is
What are you referring to by 'this'?

Reply:
The output is `29oBRYxP4TVPTfRDPa7kty1W5Ytgv1GnFyF9juRwkMK4DxPvw8K849jRGYTRzot5eh84JS8AJMUoHTDsMZrozLcgPxRu6cfmLC27V7npiuvXhzaNbtk9c4u6YoLnHMsPknFA3JZJK5XbH8tnNampKALr3Ehik1ensiy23TpgNHaXehpcNj4MbucS97BgxNdq1QbAXBz1nJABFrb6wfqot8eKFWJHJhWfRx5C5JBBTuUBhS5AtPXUoVJbUznoy21KXX1oKgRjQPy555CjqHtYkw68TfR2C3K54X6uSJY5KLjQpQgtuC3xiNAdrWaEUD3EU6wfBQ93yha3mN2vyUvFxB2oQzhv3HpYu83RiwZHQGTbEif1PD9DH6dkVxydgZgHu8ywRs4E38C` in your example, which I think is a base58-check encoded transaction. I'm not sure which endpoints we have available to consume this; for example, it may make more sense to give the option of emitting just a signature in the format expected by GraphQL, if that's how these transactions are intended to be consumed.

You could also add a flag that lets a user select their desired output format, if there are multiple different ones that may be useful.

Reply:
That is correct - the output is a base58-encoded transaction.

>I'm not sure which endpoints we have available to consume this

Perhaps it would be worthwhile to add a CLI command that can broadcast a transaction.

Reply:
The output is now the signature only, in field+scalar form, as in the GraphQL API.